### PR TITLE
Schedule XCMP task with foreign token

### PR DESCRIPTION
**Pull request by imstar15**

### CODE REVIEW

File: node/src/chain_spec/turing.rs [hunk 1]
```diff
@@ -91,42 +91,12 @@ pub fn turing_development_config() -> ChainSpec {
...
-						false,
+						Option::<VersionedMultiLocation>::encode(&Some(
+							MultiLocation::new(1, X2(Parachain(1000), PalletInstance(3))).into(),
+						)),
 						10_000_000_000_000_000_000,
 						1_000_000_000,
 						XcmFlow::Alternate,
-						Option::<VersionedMultiLocation>::encode(&Some(
-							MultiLocation::new(0, X1(PalletInstance(3))).into(),
-						)),
```
[the remark this answers is not in the document]
Comment (imstar15) [reply node/src/chain_spec/turing.rs]: This structure is outdated.

File: pallets/automation-time/src/migrations/upgrade_xcmp_task_struct.rs [hunk 2]
```diff
@@ -16,28 +15,17 @@ use xcm::latest::prelude::*;
...
-			},
-			_ => Schedule::Fixed {
-				execution_times: task.execution_times.to_vec(),
-				executions_left: task.executions_left,
-			},
-		};
 		TaskOf::<T> {
 			owner_id: task.owner_id,
 			provided_id: task.provided_id,
+			schedule: task.schedule,
```
Comment (chrisli30) [inline pallets/automation-time/src/migrations/upgrade_xcmp_task_struct.rs:27]: What code change made this query obselete?

**Review by chrisli30 — COMMENTED** [review]
Double checking, we don’t need the changes in #348 right?

https://github.com/OAK-Foundation/OAK-blockchain/commit/7d449849d9ee89596dc1ac367b813f5145b19234

File: pallets/automation-time/src/migrations/upgrade_xcmp_task_struct.rs (hunk 2 above)
Comment (chrisli30) [reply pallets/automation-time/src/migrations/upgrade_xcmp_task_struct.rs:27]: We don’t need the data transformation of schedule any more.

**Review by chrisli30 — APPROVED** [review]
End-to-end tests for task queue migration have been successfully executed on the following scenarios, with all results returning positive.

1. Shibuya Auto-staking
2. Mangata Auto-compound
3. TUR Auto-staking (scheduleAutoCompoundDelegatedStakeTask)
4. Dynamic Dispatch with Transfer 

The tests are documented in the below Google Docs(require access)
https://docs.google.com/document/d/10K9NyO-apWcVUGc2jhuvkSg3eko3hw_pPaP3pwCZyLw/edit

We are good to merge.

**Pull request merged**